### ISSUE TITLE
installation waits long time at missing blacklist URL which 404s.

### ISSUE THREAD

**Issue by TheOtherDoctor** [pi-hole/pi-hole#3440]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [ ] The issue I am reporting can be *replicated*. 
I expect so, but I did not take effort to do, see below.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)). 
Reference to existing issue linked below.

**How familiar are you with the the source code relevant to this issue?:**

0

---
**Expected behaviour:**

Installation (in my case: update to 5.0 via 'pihole -up') should be running through without extra pause - when a list source returns a 404 quickly, installation should go on right away.

**Actual behaviour:**

Gets stuck for quite some long time when trying to download https://hosts-file.net/ad_servers.txt, which is known to have disappeared and options already discussed (and finally fixed, as far as I understand) [here](https://github.com/pi-hole/pi-hole/pull/3236).

**Steps to reproduce:**

Run "pihole -up". I cannot treplicate behaviour with reasonable effort as that would involve installing a old pihole and update again.

**Troubleshooting undertaken, and/or other relevant information:**

none yet. As this is not a stopper, just a delay, ok for me now - just consider fixing for future updates please.

**Comment by dschaper** [comment]
We have removed the list from future installs. Existing installs and updates will have to remove the list manually. Pi-hole doesn’t modify a users lists because we don’t feel that should be an action we take.

Changing as users list without their knowledge or consent is not a good optioning our view.

**Comment by TheOtherDoctor** [comment]
Perfectly agree on keeping previously config'd items. My original formulation might be imprecise - I did not expect this to be auto-removed, just have the upgrade proceed quickly and possibly drop a note for the "simple" user. 

I'd think I'm not alone with this "background": when I initially installed PiHole, I got a starting set of lists somehow, and have no idea about the individual data sources. Thus, after the upgrade, a closing note like "if list source xyz stays offline, you might want to consider removing it from your config" would guide on how to proceed.

Confusing part for me has been just that the upgrade process gets stuck there for some time (subjectively...), which made me worry if something went wrong. As the Pihole upgrade process is different from standard "apt upgrade", I might have been just too unsure + impatient. 

So in the end: I just wondered that a 404 does not get handled in zero time - and a friendly summary on how to clean this up would be nice.

**Comment by PromoFaux** [comment]
I just tried a `pihole -g` run with this domain in the blocklist a few times in a row to see what sort of wait time we were looking at.

I ran it 6 or 7 times in a row and it seemed like a trivial (milliseconds) amount of time. So I figured I would record it to show that it shouldn't be an issue. Of course, as soon as I hit record and then ran `pihole -g` this time, it took maybe 30 seconds to time out...

**Comment by dschaper** [comment]
If the 404 is returned immediately then we will handle it that way. It looks like the delay is actually on the remote webservers end. `curl -I https://hosts-file.net/ad_servers.txt` takes a long time to return a 404.

**Comment by dschaper** [comment]
```plain
time curl -I https://hosts-file.net/ad_servers.txt
HTTP/1.1 404 Not Found
Date: Mon, 25 May 2020 20:47:08 GMT
Server: Apache/2.4.41 () OpenSSL/1.0.2k-fips
Content-Type: text/html; charset=iso-8859-1


real    0m5.244s
user    0m0.012s
sys     0m0.003s

time curl -I https://hosts-file.net/ad_servers.txt
HTTP/1.1 404 Not Found
Date: Mon, 25 May 2020 20:48:25 GMT
Server: Apache/2.4.41 () OpenSSL/1.0.2k-fips
Content-Type: text/html; charset=iso-8859-1


real    1m10.354s
user    0m0.018s
sys     0m0.000s

```

**Comment by TheOtherDoctor** [comment]
Ok, I see, thanks for your investigation. When I tried a similar wget (after seeing the upgrade taking the nap before the 404) I only got such quick replies and suspected the upgrade to do something extra. :-(

Sorry for bothering you with a server-side issue - especially on a data source server which we have already removed for future installs. ;-)

Probably we can consider this closed then? - To me it looks like there is nothing reasonable left to do on this thing. BTW: thanks for all your work+support, this piece of software is great.
